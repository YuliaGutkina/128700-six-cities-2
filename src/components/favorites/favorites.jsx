import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import classNames from "classnames";

import Header from "../header/header";
import {receiveUserDataSelector} from "../../reducer/user/selectors";
import {receiveFavorite} from "../../reducer/data/selectors";
import withActiveItem from "../../hocs/with-active-item/with-active-item";
import {OffersList} from "../offers-list/offers-list";


const OffersListWrapped = withActiveItem(OffersList);

const Favorites = (props) => {
  const {favorite} = props;

  return <div className={classNames(`page`, {
    "page--favorites": favorite.length,
    "page--favorites-empty": !favorite.length
  })}>
    <Header/>
    <main className={classNames(`page__main page__main--favorites`, {
      "page__main--favorites-empty": !favorite.length
    })}>
      <div className="page__favorites-container container">
        {favorite.length ?
          <section className="favorites">
            <h1 className="favorites__title">Saved listing</h1>
            <ul className="favorites__list">
              {favorite.map((cityGroup) => <li key={cityGroup.city} className="favorites__locations-items">
                <div className="favorites__locations locations locations--current">
                  <div className="locations__item">
                    <Link className="locations__item-link" to="/">
                      <span>{cityGroup.city}</span>
                    </Link>
                  </div>
                </div>
                <div className="favorites__places">
                  <OffersListWrapped
                    places={cityGroup.offers}
                    cardClassName="favorites__card"
                    imageWrapperClassName="favorites__image-wrapper"
                    cardInfoClassName="favorites__card-info"
                    imageWidth={150}
                    imageHeight={110}
                  />
                </div>
              </li>
              )}
            </ul>
          </section>
          :
          <section className="favorites favorites--empty">
            <h1 className="visually-hidden">Favorites (empty)</h1>
            <div className="favorites__status-wrapper">
              <b className="favorites__status">Nothing yet saved.</b>
              <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
            </div>
          </section>
        }
      </div>
    </main>
    <footer className="footer">
      <Link className="footer__logo-link" to="/">
        <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width={64} height={33} />
      </Link>
    </footer>
  </div>;
};

Favorites.propTypes = {
  favorite: PropTypes.arrayOf(PropTypes.object),
  userData: PropTypes.object
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  userData: receiveUserDataSelector(state),
  favorite: receiveFavorite(state)
});


export {Favorites};
export default connect(mapStateToProps)(Favorites);
