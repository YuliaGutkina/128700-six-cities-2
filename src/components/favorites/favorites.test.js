import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {Favorites} from "./favorites";


it(`Favorites page correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer
    .render(
        <Favorites/>);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
