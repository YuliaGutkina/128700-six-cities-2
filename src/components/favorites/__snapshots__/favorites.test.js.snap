// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page correctly renders after relaunch 1`] = `
<div
  className="page page--favorites"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Tel Aviv",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Paris",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <ForwardRef
                  className="locations__item-link"
                  to="/"
                >
                  <span>
                    Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Paris",
                    }
                  </span>
                </ForwardRef>
              </div>
            </div>
            <OffersList
              cardClassName="favorites__card"
              cardInfoClassName="favorites__card-info"
              className="favorites__places"
              imageHeight={110}
              imageWidth={150}
              imageWrapperClassName="favorites__image-wrapper"
            />
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <ForwardRef
      className="footer__logo-link"
      to="/"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height={33}
        src="/img/logo.svg"
        width={64}
      />
    </ForwardRef>
  </footer>
</div>
`;
